src/address.rs

```diff
@@ -129,40 +119,21 @@ pub(crate) async fn get_addresses(account: &Account, count: usize) -> crate::Res
 
 /// Generates a checksum for the given address
 // TODO: maybe this should be part of the crypto lib
-pub(crate) fn generate_checksum(address: &IotaAddress) -> crate::Result<TritBuf> {
-    let mut kerl = Kerl::new();
-    let mut hash = kerl
-        .digest(address.to_inner())
-        .map_err(|e| anyhow::anyhow!("Erro hashing the address"))?;
-    let mut trits = vec![];
-
-    for _ in 1..10 {
-        if let Some(trit) = hash.pop() {
-            trits.push(trit);
-        } else {
-            return Err(anyhow::anyhow!("Hash error"));
-        }
-    }
-
-    Ok(TritBuf::from_trits(&trits[..]))
+pub(crate) fn generate_checksum(address: &IotaAddress) -> crate::Result<String> {
```

Comment:
I think there's no need for the checksum. We are using bech32 address string from now on. This can be generated from method of `bee_transaction::Address`.

Reply:
hmm so bech32 doesn't need address checksum because the chance of it being incorrect is minimal? cc @laumair

Reply:
> hmm so bech32 doesn't need address checksum because the chance of it being incorrect is minimal?

@lucasfernog @wusyong  I am not sure I follow. Bech32 addresses do have a checksum for the HRP (Human Readable Part).